app/(gcforms)/[locale]/(form administration)/form-builder/[id]/edit/translate/components/DownloadCSVWithGroups.tsx

```diff
@@ -0,0 +1,142 @@
+import { FormElement } from "@lib/types";
+import { getDate, slugify } from "@lib/client/clientHelpers";
+import { alphabet } from "@lib/utils/form-builder";
+import { sortGroup } from "@lib/utils/form-builder/groupedFormHelpers";
+
+const formatText = (str: string) => `"${str}"`;
+
+export const DownloadCSVWithGroups = () => {
+  const { form, name } = useTemplateStore((s) => ({ form: s.form, name: s.name }));
+  const { t, i18n } = useTranslation("form-builder");
+
+  let elementIndex = 0;
+  let data = [];
+
+  const parseElement = (element: FormElement, index: string | number) => {
+    const description =
+      element.type === "richText"
+        ? formatText("Page text/Texte de page")
+        : formatText(`Question ${index}`);
+
+    if (element.type === "dynamicRow") {
+      let subElementIndex = -1;
+      data.push([
+        description,
+        formatText(element.properties.titleEn),
+        formatText(element.properties.titleFr),
+      ]);
+
+      element.properties.subElements?.map((subElement) => {
+        subElementIndex++;
+        parseElement(subElement, alphabet[subElementIndex]);
+      });
+
+      return;
+    }
+    if (element.properties.titleEn || element.properties.titleFr) {
+      data.push([
+        description,
+        formatText(element.properties.titleEn),
+        formatText(element.properties.titleFr),
+      ]);
+    }
+
+    if (element.properties.descriptionEn || element.properties.descriptionFr) {
+      data.push([
+        description,
+        formatText(element.properties.descriptionEn ?? ""),
+        formatText(element.properties.descriptionFr ?? ""),
+      ]);
+    }
+
+    if (element.properties.choices) {
+      element.properties.choices.map((choice, choiceIndex) => {
+        if (choice.en || choice.fr) {
+          data.push([`${description} - Option ${choiceIndex + 1}`, choice.en, choice.fr]);
+        }
+      });
+    }
+  };
+
+  const generateCSV = async () => {
+    data = [
+      [formatText("Description"), formatText("English/Anglais"), formatText("French/Français")],
+    ];
+    data.push([formatText("Title/Titre"), formatText(form.titleEn), formatText(form.titleFr)]);
+    data.push([
+      formatText("Description"),
+      formatText(form.introduction?.descriptionEn ?? ""),
+      formatText(form.introduction?.descriptionFr ?? ""),
+    ]);
+
+    // Sort thorugh the groups....
+    if (form.groups) {
+      const groups = form.groups;
+      Object.keys(groups).map((groupKey, index) => {
+        const group = groups[groupKey];
+        const groupElements = sortGroup({ form, group });
+
+        data.push([
+          formatText(`Group ${index + 1}`),
+          formatText(group.titleEn),
+          formatText(group.titleFr),
+        ]);
+
+        groupElements.map((element) => {
+          elementIndex++;
+          parseElement(element, elementIndex);
+        });
+      });
+    } else {
+      // If the form doesn't have groups, use the old method...
```

Comment:
We can remove this 

All entry points for "group" wrappers already.

```
export default async function Page({ params: { id } }: { params: { id: string } }) {
  const conditionalLogic = await allowGrouping();

  return (
    <>
      {!conditionalLogic && <EditNavigation id={id} />}
      {conditionalLogic ? <TranslateWithGroups /> : <Translate />}
    </>
  );
}
```

Same sort of thing for the right panels